### PR TITLE
Rest api v0.5 store file name from url

### PR DESCRIPTION
**Pull request by ZNick1982**
This pr resolves issue #79 

Changes contains:
- Added new input variable `$upload_params` with dictionary of method parameters.
- Moved dedicated parameters to $upload_params dictionary + added deprecated messages if use it.
- Fixed float CI error with 404 after upload text file.

### CODE REVIEW

File: src/Uploadcare/Uploader.php [hunk 1]
```diff
@@ -51,16 +51,64 @@ public function status($token)
...
+   * 'store' - can be true, false or 'auto'. This flag indicates should file be stored automatically after upload.
+   * 'filename' - should be a string, Sets explicitly file name of uploaded file.
+   * 'check_status' - Wait till upload is complete
+   * 'timeout' - Wait number of seconds between status checks
+   * 'max_attempts' - Check status no more than passed number of times
    * @return File|string
    * @throws \Exception
    */
-  public function fromUrl($url, $check_status = true, $timeout = 1, $max_attempts = 5)
+  public function fromUrl($url, $check_status = true, $timeout = 1, $max_attempts = 5, $upload_params = array())
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Uploader.php]: $upload_params -> $options

File: src/Uploadcare/Uploader.php [hunk 2]
```diff
@@ -51,16 +51,64 @@ public function status($token)
    * @param boolean $check_status Wait till upload is complete
    * @param int $timeout Wait $timeout seconds between status checks
    * @param int $max_attempts Check status no more than $max_attempts times
+   * @param array $upload_params Optioanal dictionary with additional params. Available keys are follwing:
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Uploader.php]: typo

File: src/Uploadcare/Uploader.php [hunk 3]
```diff
@@ -51,16 +51,64 @@ public function status($token)
    * @param boolean $check_status Wait till upload is complete
    * @param int $timeout Wait $timeout seconds between status checks
    * @param int $max_attempts Check status no more than $max_attempts times
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Uploader.php]: mark all these as deprecated in docstring

File: src/Uploadcare/Uploader.php [hunk 4]
```diff
@@ -51,16 +51,64 @@ public function status($token)
...
+   * 'check_status' - Wait till upload is complete
+   * 'timeout' - Wait number of seconds between status checks
+   * 'max_attempts' - Check status no more than passed number of times
    * @return File|string
    * @throws \Exception
    */
-  public function fromUrl($url, $check_status = true, $timeout = 1, $max_attempts = 5)
+  public function fromUrl($url, $check_status = true, $timeout = 1, $max_attempts = 5, $upload_params = array())
   {
+    $defParams = array(
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Uploader.php]: $default_options

File: src/Uploadcare/Uploader.php (hunk 3 above)
Comment (ZNick1982) [inline src/Uploadcare/Uploader.php]: done

File: src/Uploadcare/Uploader.php (hunk 2 above)
Comment (ZNick1982) [inline src/Uploadcare/Uploader.php]: fixed

File: src/Uploadcare/Uploader.php (hunk 1 above)
Comment (ZNick1982) [inline src/Uploadcare/Uploader.php]: done

File: tests/ApiTest.php [hunk 5]
```diff
@@ -279,12 +279,30 @@ public function testCustomCDNHost()
    */
   public function testUploadFromURL()
   {
+    $params = array(
+      'filename' => 'IMG_1.jpg',
+      'store' => true,
+    );
     try {
-      $file = $this->api->uploader->fromUrl('https://www.baysflowers.co.nz/wp-content/uploads/2015/06/IMG_9886_2.jpg');
+      $file = $this->api->uploader->fromUrl('https://www.baysflowers.co.nz/wp-content/uploads/2015/06/IMG_9886_2.jpg', true, 1, 5, $params);
```
Comment (dmitry-mukhin) [inline tests/ApiTest.php]: use new $options?

File: tests/ApiTest.php [hunk 6]
```diff
@@ -279,12 +279,30 @@ public function testCustomCDNHost()
...
+    $data = $file->__get('data');
+    $fileName = $data["original_filename"];
     $this->assertEquals(get_class($file), 'Uploadcare\File');
+    $this->assertEquals($fileName, $params['filename']);
+    $this->assertTrue(!!$data["datetime_stored"]);
+    $file->delete();
+
+    try {
+      $params['store'] = false;
+      $file = $this->api->uploader->fromUrl('https://www.baysflowers.co.nz/wp-content/uploads/2015/06/IMG_9886_2.jpg', true, 1, 5, $params);
```
Comment (dmitry-mukhin) [inline tests/ApiTest.php]: use new $options?

File: tests/ApiTest.php (hunk 5 above)
Comment (ZNick1982) [inline tests/ApiTest.php]: renamed to $options now its look more clear

File: tests/ApiTest.php [hunk 7]
```diff
@@ -279,25 +279,25 @@ public function testCustomCDNHost()
...
   public function testUploadFromURL()
   {
-    $params = array(
+    $options = array(
       'filename' => 'IMG_1.jpg',
       'store' => true,
     );
     try {
-      $file = $this->api->uploader->fromUrl('https://www.baysflowers.co.nz/wp-content/uploads/2015/06/IMG_9886_2.jpg', true, 1, 5, $params);
+      $file = $this->api->uploader->fromUrl('https://www.baysflowers.co.nz/wp-content/uploads/2015/06/IMG_9886_2.jpg', true, 1, 5, $options);
```
Comment (dmitry-mukhin) [inline tests/ApiTest.php]: I was talking about true 1 5

Comment (ZNick1982) [inline tests/ApiTest.php]: Added its also in the $options array.

File: src/Uploadcare/Uploader.php [hunk 8]
```diff
@@ -44,23 +44,82 @@ public function status($token)
...
+      else if(count($arguments) >= 3) {
+        return call_user_func_array(array($this,'fromUrlOld'), $arguments);
+      }
+    }
+  }
+
   /**
    * Upload file from url and get File instance
    *
+   * @deprecated 2.0.0 please use fromUrlNewfromUrlNew($url, $options) instead
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Uploader.php]: something happened here

**Comment by dmitry-mukhin** [comment]
great!

**Pull request merged**